src/Rules/Properties/AlwaysUsedClassConstantsExtension.php

```diff
@@ -0,0 +1,12 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Properties;
+
+use PHPStan\Reflection\ConstantReflection;
+
+interface AlwaysUsedClassConstantsExtension
+{
+
+	public function isAlwaysRead(ConstantReflection $constant, string $constantName): bool;
```

Comment:
1) The extension should ask for `ClassConstantReflection`
2) The second argument isn't needed as `$constant` has `getName()`.

Reply:
3. `isAlwaysUsed`, not `Read`

Reply:
In this comment you ask to to replace the `ConstantReflection` which is an interface by `ClassConstantReflection` which is the implementation of the interface.

Could you explain which I should use the implementation and not the interface ?

Moreover, PHPStan fails in the CI because in the `nusedPrivateConstantRule` when I call my extension, it gives n `ConstantReflection` and not a `ClassConstantReflection`.

Reply:
Oh yes, sorry, `PHPStan\Reflection\ConstantReflection` interface is the right choice here.

Reply:
Change it back to `PHPStan\Reflection\ConstantReflection`, make sure the build passes and I'll merge this 👍